### ISSUE TITLE
Publishing XLSForms with same FormId with different case shouldn't be possible

### ISSUE THREAD

**Issue by hkmshb** [onaio/onadata#782]
It is possible to publish multiple XLSForms with the same FormId if the FormId is written in different case. However both forms are unsable as the URL for operations on these forms are the same.

I created an XLSForm with FormId **'custform01_mrr'** but had to make changes to the form after it was published. I modified the XLSForm and had intended to modify the FormId to read 02 however I had only changed the FormId to **'CustForm01_MRR'**, and I was able to publish it. I only realized the mistake when I couldn't use or delete either of the forms which, as it happens, where both listed under Published Forms.

**Comment by WinnyTroy** [comment]
Looking through this issue, I believe this to be an intentional feature as opposed to being a bug.

In the event that a user tries to upload a form with an id_string/form_id/form_title `that already exists, but is of different case`, it is handled here:
https://github.com/onaio/onadata/blob/09c0bdb3cf7bb3efc3c763e929b58183f192c60e/onadata/apps/logger/models/xform.py#L286

And tested accordingly here: https://github.com/onaio/onadata/blob/fdd91d7024c9b15ee1eee4e074e2e0ef21939b8b/onadata/apps/api/tests/viewsets/test_xform_viewset.py#L1099

**Comment by pld** [comment]
OK, but it sounds like there is an error in-so-far as when you have forms with ID strings in different cases you can't use or delete them, as @hkmshb is reporting.

We can:

1. Change to ignore case in uniqueness tests
2. Fix the functionalities that are broken when equivalent-without-case-information forms exist

Are there other options, thoughts on what to do?

**Comment by WinnyTroy** [comment]
@pld I did test out those functionalities and there were no errors registered.

I was able to upload the same xls file, three times to the same project. 
Make submissions to all three forms.
Make Exports of the submission data.
Delete individual submissions from the three forms.
Delete the form entirely from the projects and re-upload the same file once again.


![same_file](https://user-images.githubusercontent.com/11174326/72433879-5e584580-37ab-11ea-9635-dcb2c35ab5b8.png)
